node/src/proxy_server/mod.rs

```diff
@@ -162,12 +164,35 @@ impl AddReturnRouteMessage {
     }
 }
 
-impl Handler<AddRouteMessage> for ProxyServer {
+impl Handler<RouteResultMessage> for ProxyServer {
     type Result = ();
 
-    fn handle(&mut self, msg: AddRouteMessage, _ctx: &mut Self::Context) -> Self::Result {
-        debug!(self.logger, "Establishing stream key {}", msg.stream_key);
-        self.stream_key_routes.insert(msg.stream_key, msg.route);
+    fn handle(&mut self, msg: RouteResultMessage, _ctx: &mut Self::Context) -> Self::Result {
+        let dns_failure = self
```

Comment:
We've been very consistent with the name `dns_failure` while implementing this card. Keeping it the same will help in maintaining consistency. Also, this variable stores the client request, which is a form of caching. Since the cached data of the dns failure is stored inside the variable, it seems more appropriate to keep the name `dns_failure` over here.